src/data_types/strs.rs

```diff
@@ -161,6 +163,44 @@ impl CStr16 {
             pos: 0,
         }
     }
+
+    /// Write a string slice into the provided buffer. If it fails, then most probably, because
+    /// the buffer is not big enough. In that case, the buffer will contain the correct string
+    /// until the point, where the size was not enough.
+    ///
+    /// ## Example
+    /// ```rust
+    /// use uefi::data_types::ArrayString;
+    /// use uefi::{CStr16, Char16};
+    /// let firmware_vendor_c16_str: CStr16 = ...;
+    /// // crate "arrayvec" uses stack-allocated arrays for Strings => no heap
+    /// let mut buf = arrayvec::ArrayString::<128>::new();
+    /// firmware_vendor_c16_str.as_str_in_buf(&mut buf);
+    /// log::info!("as rust str: {}", buf.as_str());
+    /// ```
+    pub fn as_str_in_buf(&self, buf: &mut dyn core::fmt::Write) -> Result<(), ()> {
+        for c16 in self.iter() {
+            let res = buf.write_char(char::from(*c16));
+            if let Err(err) = res {
+                log::error!("Failed to write CStr16 as &str into buffer. Buffer too small? ({})", err);
+                return Err(())
```

Comment:
Maybe we could return an error type, a custom enum defined in this module?

Reply:
I don't know it would be beneficial. The only error that can happen is that `write_char()` fails. This is most probably because the buffer was too small.. but there is no guarantee for that..

Reply:
The thing is, we generally don't want to write messages to the current logger in our library code. It should be up to the client to decide how to handle the error conditions.

`write_char` returns a [`core::fmt::Error`](https://doc.rust-lang.org/core/fmt/struct.Error.html). Could we return that in the function's result?